### PR TITLE
Radio button docs

### PR DESCRIPTION
**Pull request by britt6612**
What does this PR do?

Doc improvements to Radio Button
Where should the reviewer start?
/src/radioButton/docs
src/js/RangeSelector/doc.js
What testing has been done on this PR?
How should this be manually tested?
Any background context you want to provide?
What are the relevant issues?

#2818
Screenshots (if appropriate)
Do the grommet docs need to be updated?
auto
Should this PR be mentioned in the release notes?
no
Is this change backwards compatible or is it a breaking change?
backward compatible

### CODE REVIEW

File: src/js/components/RadioButton/doc.js [hunk 1]
```diff
@@ -40,3 +40,57 @@ with the same name so form submissions work.`,
 
   return DocumentedRadioButton;
 };
+
+export const themeDoc = {
+  'radioButton.border.color': {
+    description: 'The color of the border of the Radio Button.',
+    type: 'string',
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]:
```suggestion
    type: 'string | { dark: string, light: string }',
```

File: src/js/components/RadioButton/doc.js [hunk 2]
```diff
@@ -40,3 +40,57 @@ with the same name so form submissions work.`,
...
+      "{dark: 'rgba(255, 255, 255, 0.5), light: 'rgba(0, 0, 0, 0.15)}",
+  },
+  'radioButton.border.width': {
+    description: 'The width size of the border of the Radio Button.',
+    type: 'string',
+    defaultValue: '2px',
+  },
+  'radioButton.check.color': {
+    description: 'The color of the checked icon in the Radio Button.',
+    type: 'string',
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]:
```suggestion
    type: 'string | { dark: string, light: string }',
```

File: src/js/components/RadioButton/doc.js [hunk 3]
```diff
@@ -40,3 +40,57 @@ with the same name so form submissions work.`,
...
+    description: 'Any additional style for the Radio Button Check.',
+    type: 'string | (props) => {}',
+  },
+  'radioButton.check.radius': {
+    description: 'The radius size for the Radio Button.',
+    type: 'string',
+    defaultValue: '100%',
+  },
+  'radioButton.extend': {
+    description: 'ny additional style for the Radio Button.',
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]:
```suggestion
    description: 'Any additional style for the RadioButton.',
```

File: src/js/components/RadioButton/doc.js [hunk 4]
```diff
@@ -40,3 +40,57 @@ with the same name so form submissions work.`,
...
+    type: 'string | (props) => {}',
+  },
+  'radioButton.gap': {
+    description: 'The gap between the label and the Radio Button.',
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]:
```suggestion
    description: 'The gap between the label and the Radio Button itself.',
```

File: src/js/components/RadioButton/doc.js [hunk 5]
```diff
@@ -40,3 +40,57 @@ with the same name so form submissions work.`,
...
+    defaultValue: 'small',
+  },
+  'radioButton.size': {
+    description: 'The size of the Radio Buttons.',
+    type: 'string',
+  },
+  'radioButton.hover.border.color': {
+    description: `The color of the Radio Button border when hovered over.`,
+    type: 'string | { dark: string, light: string }',
+    defaultValue: 'dark: white, light: black',
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]:
```suggestion
    defaultValue: '{dark: white, light: black}',
```

File: src/js/components/RadioButton/doc.js [hunk 6]
```diff
@@ -40,3 +40,57 @@ with the same name so form submissions work.`,
...
+  'radioButton.extend': {
+    description: 'ny additional style for the Radio Button.',
+    type: 'string | (props) => {}',
+  },
+  'radioButton.gap': {
+    description: 'The gap between the label and the Radio Button.',
+    type: 'string',
+    defaultValue: 'small',
+  },
+  'radioButton.size': {
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]: should default to 24px.

The base.js code says:
`size: `${baseSpacing}px`,`
And on top of the file, you can see that `baseSpacing` defaults to 24.

so default value should be 24px.

Comment (britt6612) [reply src/js/components/RadioButton/doc.js]: Yes thank you for catching!

File: src/js/components/RadioButton/doc.js [hunk 7]
```diff
@@ -40,3 +40,58 @@ with the same name so form submissions work.`,
...
+
+export const themeDoc = {
+  'radioButton.border.color': {
+    description: 'The color of the border of the Radio Button.',
+    type: 'string | { dark: string, light: string }',
+    defaultValue:
+      "{dark: 'rgba(255, 255, 255, 0.5), light: 'rgba(0, 0, 0, 0.15)}",
+  },
+  'radioButton.border.width': {
+    description: 'The width size of the border of the Radio Button.',
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]:
```suggestion
    description: 'The width size of the border of the RadioButton.',
```

File: src/js/components/RadioButton/doc.js [hunk 8]
```diff
@@ -40,3 +40,58 @@ with the same name so form submissions work.`,
...
+    type: 'string',
+    defaultValue: '2px',
+  },
+  'radioButton.check.color': {
+    description: 'The color of the checked icon in the Radio Button.',
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]:
```suggestion
    description: 'The color of the checked icon in the RadioButton.',
```

File: src/js/components/RadioButton/doc.js [hunk 9]
```diff
@@ -40,3 +40,58 @@ with the same name so form submissions work.`,
...
+    type: 'string | { dark: string, light: string }',
+    defaultValue: 'string | { dark: string, light: string }',
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]: the default should be `undefined`

File: src/js/components/RadioButton/doc.js [hunk 10]
```diff
@@ -40,3 +40,58 @@ with the same name so form submissions work.`,
...
+  },
+  'radioButton.check.extend': {
+    description: 'Any additional style for the Radio Button Check.',
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]:
```suggestion
    description: 'Any additional style for the checked RadioButton.',
```

File: src/js/components/RadioButton/doc.js [hunk 11]
```diff
@@ -40,3 +40,58 @@ with the same name so form submissions work.`,
...
+    type: 'string | (props) => {}',
+  },
+  'radioButton.check.radius': {
+    description: 'The radius size for the Radio Button.',
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]:
```suggestion
    description: 'The border-radius of the RadioButton.',
```

File: src/js/components/RadioButton/doc.js [hunk 12]
```diff
@@ -40,3 +40,58 @@ with the same name so form submissions work.`,
...
+    type: 'string',
+    defaultValue: '100%',
+  },
+  'radioButton.extend': {
+    description: 'Any additional style for the RadioButton.',
+    type: 'string | (props) => {}',
+  },
+  'radioButton.gap': {
+    description: 'The gap between the label and the Radio Button itself.',
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]:
```suggestion
    description: 'The gap between the label and the RadioButton itself.',
```

File: src/js/components/RadioButton/doc.js [hunk 13]
```diff
@@ -40,3 +40,58 @@ with the same name so form submissions work.`,
...
+    type: 'string',
+    defaultValue: 'small',
+  },
+  'radioButton.size': {
+    description: 'The size of the Radio Buttons.',
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]:
```suggestion
    description: 'The size of the RadioButton.',
```

File: src/js/components/RadioButton/doc.js [hunk 14]
```diff
@@ -40,3 +40,58 @@ with the same name so form submissions work.`,
...
+    type: 'string',
+    defaultValue: '24px',
+  },
+  'radioButton.hover.border.color': {
+    description: `The color of the Radio Button border when hovered over.`,
+    type: 'string | { dark: string, light: string }',
+    defaultValue: '{dark: white, light: black}',
+  },
+  'radioButton.icon.extend': {
+    description: 'Any additional style for the Radio Button Icon.',
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]:
```suggestion
    description: 'Any additional style for the RadioButton icon.',
```

File: src/js/components/RadioButton/doc.js [hunk 15]
```diff
@@ -40,3 +40,58 @@ with the same name so form submissions work.`,
...
+    type: 'string',
+    defaultValue: 'small',
+  },
+  'radioButton.size': {
+    description: 'The size of the Radio Buttons.',
+    type: 'string',
+    defaultValue: '24px',
+  },
+  'radioButton.hover.border.color': {
+    description: `The color of the Radio Button border when hovered over.`,
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]:
```suggestion
    description: `The color of the RadioButton border when hovered over.`,
```

File: src/js/components/RadioButton/doc.js [hunk 16]
```diff
@@ -40,3 +40,58 @@ with the same name so form submissions work.`,
...
+    type: 'string | { dark: string, light: string }',
+    defaultValue: '{dark: white, light: black}',
+  },
+  'radioButton.icon.extend': {
+    description: 'Any additional style for the Radio Button Icon.',
+    type: 'string | (props) => {}',
+  },
+  'radioButton.icon.size': {
+    description: 'The size of the icon in the Radio Button.',
```
Comment (ShimiSun) [inline src/js/components/RadioButton/doc.js]:
```suggestion
    description: 'The size of the icon in the RadioButton.',
```

**Pull request merged**